src/models/auth.js

```diff
@@ -87,9 +98,68 @@ export default {
         payload,
       });
     },
+    *removeErrorMessage({ payload }, { put }) {
+      yield put({
+        type: 'modifyErrorMessage',
+        payload,
+      });
+    },
+    *removeSuccessMessage({ payload }, { put }) {
+      yield put({
+        type: 'modifySuccessMessage',
+        payload,
+      });
+    },
   },
 
   reducers: {
+    LOGIN_USER_SUCCESS(state, { payload }) {
+      const { username } = payload;
+      return {
+        ...state,
+        errorMessage: '',
+        username,
+      };
+    },
+    LOGIN_USER_FAILURE(state, { payload }) {
+      return {
+        ...state,
+        errorMessage: payload,
+      };
+    },
+    REGISTER_USER_SUCCESS(state, { payload }) {
+      return {
+        ...state,
+        successMessage: payload,
+        errorMessage: '',
+      };
+    },
+    REGISTER_USER_FAILURE(state, { payload }) {
+      return {
+        ...state,
+        errorMessage: payload,
+        successMessage: '',
+      };
+    },
+    LOGOUT_USER_SUCCESS(state) {
+      return {
+        ...state,
+        username: '',
+        errorMessage: '',
+        successMessage: '',
+      };
+    },
+    // A failure in logout also removes
+    // the user. We do not want users to be
```

Comment:
"removes the user" implies that we're deleting the user rather than merely removing the logged-in username from the Redux state.

Reply:
fixed it, thanks!